FABulous.py

```diff
@@ -0,0 +1,743 @@
+            self.fabricGenerator = FabricGenerator(self.fabric, self.writer)
+
+        if isinstance(self.writer, VHDLWriter):
+            self.fileExtension = ".vhdl"
+        # self.fabricGenerator = FabricGenerator(fabric, writer)
+
+    def setWriterOutputFile(self, outputDir):
+        self.writer.outFileName = outputDir
+
+    def loadFabric(self, dir: str):
+        if dir.endswith(".csv"):
+            self.fabric = fileParser.parseFabricCSV(dir)
+            self.fabricGenerator = FabricGenerator(self.fabric, self.writer)
+        else:
+            logger.warning("Only .csv files are supported for fabric loading")
+
+    def bootstrapSwitchMatrix(self, tileName: str, outputDir: str):
+        tile = self.fabric.getTileByName(tileName)
+        self.fabricGenerator.bootstrapSwitchMatrix(tile, outputDir)
+
+    def addList2Matrix(self, list: str, matrix: str):
+        self.fabricGenerator.list2CSV(list, matrix)
+
+    def genConfigMem(self, tileName: str, configMem: str):
+        tile = self.fabric.getTileByName(tileName)
+        self.fabricGenerator.generateConfigMem(tile, configMem)
+
+    def genSwitchMatrix(self, tileName: str):
+        tile = self.fabric.getTileByName(tileName)
+        self.fabricGenerator.genTileSwitchMatrix(tile)
+
+    def genTile(self, tileName: str):
+        tile = self.fabric.getTileByName(tileName)
+        self.fabricGenerator.generateTile(tile)
+
+    def genSuperTile(self, tileName: str):
+        tile = self.fabric.getSuperTileByName(tileName)
+        self.fabricGenerator.generateSuperTile(tile)
+
+    def genFabric(self):
+        self.fabricGenerator.generateFabric()
+
+    def genTopWrapper(self):
+        self.fabricGenerator.generateTopWrapper()
+
+    def genBitStreamSpec(self):
+        specObject = self.fabricGenerator.generateBitsStreamSpec()
+        return specObject
+
+
+class FABulousShell(cmd.Cmd):
+    intro: str = f"""
+
+    ______      ____        _
+    |  ____/\   |  _ \      | |
+    | |__ /  \  | |_) |_   _| | ___  _   _ ___
+    |  __/ /\ \ |  _ <| | | | |/ _ \| | | / __|
+    | | / ____ \| |_) | |_| | | (_) | |_| \__ \\
+    |_|/_/    \_\____/ \__,_|_|\___/ \__,_|___/
+
+
+Welcome to FABulous shell
+You have started FABlous shell with following options:
+{' '.join(sys.argv[1:])}
+
+Type help or ? to list commands
+The shell support tab completion for commands and files
+
+To run the complete FABulous flow, type 
+    load_fabric <fabric_file>
+    run_FABulous
+    """
+    prompt: str = "FABulous> "
+    fabricGen: FABulous
+    projectDir: str
+    top: str
+    allTile: List[str]
+    csvFile: str
+    extension: str = "v"
+    fabricLoaded: bool = False
+
+    def __init__(self, fab: FABulous, projectDir: str):
+        super().__init__()
+        self.fabricGen = fab
+        self.projectDir = projectDir
+        self.tiles = []
+        self.superTiles = []
+        self.csvFile = ""
+        self.allTile = [f.name for f in os.scandir(f"./Tile") if f.is_dir()]
```

Comment:
this seems to be causing a failure for me. there's a `Tile/CPU_IO` subfolder in the example project but no reference to `CPU_IO` in `fabric.csv`, causing this error:

```
  File "/home/gatecat/FABulous-refactor/FABulous.py", line 343, in do_gen_fabric
    self.do_gen_all_tile()
  File "/home/gatecat/FABulous-refactor/FABulous.py", line 337, in do_gen_all_tile
    self.do_gen_tile(self.allTile)
  File "/home/gatecat/FABulous-refactor/FABulous.py", line 317, in do_gen_tile
    self.do_gen_switch_matrix(t)
  File "/home/gatecat/FABulous-refactor/FABulous.py", line 261, in do_gen_switch_matrix
    self.fabricGen.genSwitchMatrix(i)
  File "/home/gatecat/FABulous-refactor/FABulous.py", line 106, in genSwitchMatrix
    tile = self.fabric.getTileByName(tileName)
  File "/home/gatecat/FABulous-refactor/fabric_generator/fabric.py", line 439, in getTileByName
    return self.tileDic[name]
KeyError: 'CPU_IO'
```

I think this should probably be populated later based on `fabric.csv` contents rather than just a directory enumeration?